python/ray/util/client/server/proxier.py

```diff
@@ -0,0 +1,293 @@
+from concurrent import futures
+from dataclasses import dataclass
+import grpc
+import logging
+import json
+from queue import Queue
+import random
+import socket
+from threading import Thread, Lock
+import time
+from typing import Any, Callable, Dict, Iterator, Optional
+
+import ray.core.generated.ray_client_pb2 as ray_client_pb2
+import ray.core.generated.ray_client_pb2_grpc as ray_client_pb2_grpc
+from ray.util.client.common import (ClientServerHandle,
+                                    CLIENT_SERVER_MAX_THREADS, GRPC_OPTIONS)
+from ray._private.services import ProcessInfo, start_ray_client_server
+
+logger = logging.getLogger(__name__)
+
+CHECK_PROCESS_INTERVAL_S = 30
+
+MIN_SPECIFIC_SERVER_PORT = 23000
+
+MAX_SPECIFIC_SERVER_PORT = 24000
+
+
+def _get_unused_port() -> int:
+    for _ in range(10):
+        port = random.randint(MIN_SPECIFIC_SERVER_PORT,
+                              MAX_SPECIFIC_SERVER_PORT)
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        try:
+            s.bind(("", port))
+        except OSError:
+            s.close()
+            continue
+        s.close()
+        s.close()
+        return port
+    raise RuntimeError("Unable to succeed in selecting a random port.")
```

Comment:
how about bind with 0 like [here](https://stackoverflow.com/questions/1365265/on-localhost-how-do-i-pick-a-free-port-number)

Reply:
@iycheng I want to bind to a free port in this range. The reason is that I don't want to choose a port in one of the other ranges used by ray.